app/src/main/java/com/darkempire78/opencalculator/MainActivity.kt

```diff
@@ -393,54 +420,59 @@ class MainActivity : AppCompatActivity() {
     }
 
     fun equalsButton(view: View) {
-        keyVibration(view)
 
-        var calculation = input.text.toString()
-        calculation = calculation.replace('×', '*')
-        calculation = calculation.replace('÷', '/')
-        calculation = calculation.replace("log", "log10")
+        lifecycleScope.launch(Dispatchers.Default) {
+            keyVibration(view)
 
-        if (calculation != "") {
-            // Add ")" which lack
-            var openParentheses = 0
-            var closeParentheses = 0
+            var calculation = input.text.toString()
+            calculation = calculation.replace('×', '*')
+            calculation = calculation.replace('÷', '/')
+            calculation = calculation.replace("log", "log10")
 
-            for (i in 0..calculation.length-1) {
-                if (calculation[i] == '(') {
-                    openParentheses += 1
-                }
-                if (calculation[i] == ')') {
-                    closeParentheses += 1
+            if (calculation != "") {
+                // Add ")" which lack
+                var openParentheses = 0
+                var closeParentheses = 0
+
+                for (i in 0..calculation.length - 1) {
+                    if (calculation[i] == '(') {
+                        openParentheses += 1
+                    }
+                    if (calculation[i] == ')') {
+                        closeParentheses += 1
+                    }
                 }
-            }
-            if (closeParentheses < openParentheses) {
-                for (i in 0..openParentheses-closeParentheses-1) {
-                    calculation += ')'
+                if (closeParentheses < openParentheses) {
+                    for (i in 0..openParentheses - closeParentheses - 1) {
+                        calculation += ')'
+                    }
                 }
-            }
 
-            val exp = Expression(calculation)
-            var result = exp.calculate().toString()
+                val exp = Expression(calculation)
+                var result = exp.calculate().toString()
+
+                mXparser.consolePrintln("Res: " + exp.expressionString.toString() + " = " + exp.calculate())
 
-            mXparser.consolePrintln("Res: " + exp.expressionString.toString() + " = " + exp.calculate())
+                if (result != "NaN" && result != "Infinity") {
+                    if ((exp.calculate() * 10) % 10 == 0.0) {
```

Comment:
I'd also use a single `withContext` block here for the whole if-else block and move the `calculate`-call up.

Reply:
Thank you for your suggestion, I have made these changes as per your suggestion.